### PR TITLE
feat(allure-jest): Add ability to assign custom names for any attachments (feat #801)

### PR DESCRIPTION
**Pull request by Louis-Wu-Sea**
### Context

Add ability to assign custom names for any attachments #801

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: packages/allure-jest/src/AllureJestApi.ts [hunk 1]
```diff
@@ -59,13 +59,13 @@ export class AllureJestApi implements AllureRuntimeApiInterface {
...
 
-  attachment(content: string | Buffer, type: string): void {
+  attachment(content: string | Buffer, type: string, name?: string): void {
     const isBuffer = Buffer.isBuffer(content);
 
     this.sendMetadata({
       attachments: [
         {
-          name: "Attachment",
+          name: name ? name : "Attachment",
```
Comment (epszaw) [inline packages/allure-jest/src/AllureJestApi.ts:68]: Let's make it shorter `name || "Attachment"`

File: packages/allure-jest/test/fixtures/attachments.test.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
 it("json", () => {
   allure.attachment(JSON.stringify({ foo: "bar" }), "application/json");
 });
+
+it("name", () => {
```
Comment (epszaw) [inline packages/allure-jest/test/fixtures/attachments.test.js:5]: `it("with name", () => {`, now it looks like we have method called `name`, not an ability to set the property

**Review by epszaw — CHANGES_REQUESTED** [review]
Good job there! Please fix some moments I've highlighted in the review to continue

File: packages/allure-jest/src/AllureJestApi.ts (hunk 1 above)
Comment (Louis-Wu-Sea) [reply packages/allure-jest/src/AllureJestApi.ts]: done

File: packages/allure-jest/test/fixtures/attachments.test.js (hunk 2 above)
Comment (Louis-Wu-Sea) [reply packages/allure-jest/test/fixtures/attachments.test.js]: done

**Review by baev — APPROVED** [review]

**Comment by Louis-Wu-Sea** [comment]
@baev @epszaw I have extended the "attachment" method for jest only. Do I need to add for other frameworks?

**Comment by baev** [comment]
@Louis-Wu-Sea, it would be nice. We can do other integrations in separate PR since some may be not that straightforward.

**Review by epszaw — APPROVED** [review]

**Comment by baev** [comment]
@Louis-Wu-Sea thanks

**Comment by Louis-Wu-Sea** [comment]
@baev @epszaw How often do you release packages? So I know when I can use it?

**Comment by baev** [comment]
I'll try to release it today.

**Pull request merged**